pkg/kubelet/cm/dra/claiminfo.go

```diff
@@ -118,10 +158,32 @@ func (cache *claimInfoCache) hasPodReference(UID types.UID) bool {
 	defer cache.RUnlock()
 
 	for _, claimInfo := range cache.claimInfo {
-		if claimInfo.podUIDs.Has(string(UID)) {
+		if claimInfo.PodUIDs.Has(string(UID)) {
 			return true
 		}
 	}
 
 	return false
 }
+
+func (cache *claimInfoCache) addPodReference(claimName, namespace string, podUID types.UID) {
+	cache.Lock()
+	defer cache.Unlock()
+
+	claimInfo := cache.claimInfo[claimName+namespace]
+	claimInfo.PodUIDs.Insert(string(podUID))
+
+	claimInfoStateList := cacheToState(cache.claimInfo)
+	cache.state.Store(claimInfoStateList)
```

Comment:
Do we really want to be storing this back to the checkpoint file every time we update the pod reference? I would think we need to think carefully about when we sync info back to the checkpoint file so we don't end up in some weird intermediate state if something fails after we have updated this reference.

Reply:
after rethinking this I agree there is no point of checkpointing on addPodReference or deletePodReference.
 
So it seem we can either save on add and delete or save after we run  over on all the containers resource claims. 


[1] https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/dra/manager.go#L241-L245
[2] - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/dra/manager.go#L146

Reply:
I change it to checkpoint on the end of PrepareResources and UnprepareResources. 

The only problem I found with the old code is that on UnprepareResources we delete the last pod ref and than delete the claim [1]. (On add we are doing it in the same call [2]).  add and delete are a bit inconsistent.

The only reason it was implemented to store on every call operation(add/delete/add pod red/delete pod ref) was to reduce the unnecessary call to DRA plugins when kubelet restart but not finished to go over on all the Pod resources. 
maybe because DRA plugins should be idempotent so that not an issue. 

[1] - https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/dra/manager.go#L241-L245 
[2] -https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/cm/dra/manager.go#L146